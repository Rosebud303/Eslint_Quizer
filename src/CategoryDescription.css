.definition-image {
    width: 300px;
    height: 300px;
    background: red;
}

.description-card {
    position: absolute;
    width: 300px;
    background: #8bc34a;
    padding: 30px;
    border-radius: 25px;
    border: 5px solid #2d9c5f;
    box-shadow: 4px 4px 20px black;
}

.card-name {
    text-align: center;
    margin-top: 0;
}

.card-description {
    background: #bde4e2;
    padding: 10px;
    border-radius: 10px;
    border: 4px solid;
}

button {
    display: block;
    margin: auto;
    width: 150px;
    height: 30px;
    margin-top: 40px;
    border-radius: 8px;
    border: 3px solid;
    background: #009688;
}

button:hover {
    background: #2196f3;
    cursor: pointer;
}