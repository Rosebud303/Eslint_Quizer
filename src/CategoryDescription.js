import React, { Component } from 'react';
import './CategoryDescription.css';

export default class CategoryDescription extends Component {
    
  render() {
    const { name, description, url, correctImg, incorrectImg} = this.props.selectedCat
    
    return (
      <div className='description-card'>
        <h3 className='card-name'>{name}</h3>
        <p className='card-description'>{description}</p>
        <p>This is how not to do it:</p>
        <img className='category-image' src={incorrectImg}/>
        <p>This is the proper way to do it:</p>
        <img className='category-image' src={correctImg}/>
        <p>Link for more info:</p>
        <a href={url}>{url}</a>
        <button className='return-button' onClick={this.props.toggle}>Finished</button>
      </div>
    )
  }
}
