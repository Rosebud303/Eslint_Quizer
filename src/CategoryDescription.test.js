import React from 'react';
import CategoryDescription from './CategoryDescription.js';
import { shallow } from 'enzyme';

const mockCata =     {
    "name": "Indentation",
    "description": "When refering to 'Indentation' in esLint this means we are talking about the stylistic way your code should look in the developer. Indentation makes your code more readable and is the first step into writing less spaghetti code. It also gives the reader visual ques for hierarchy and scoping.",
    "incorrectImg": "https://i.imgur.com/oR17Hke.png",
    "correctImg": "https://i.imgur.com/YIp5OaH.png",
    "url": "https://eslint.org/docs/rules/indent"
};

const mockFunc = jest.fn();

describe('CategoryDescription', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <CategoryDescription selectedCat={mockCata}
                                 toggle={mockFunc}
            />
        )
    });

    it('Should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should simulate a click', () => {
        wrapper.find('.return-button').simulate('click');
        expect(mockFunc).toHaveBeenCalled();
    });
});