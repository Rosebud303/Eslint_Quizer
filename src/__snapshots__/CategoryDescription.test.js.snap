// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryDescription Should match the snapshot 1`] = `
<div
  className="description-card"
>
  <h3
    className="card-name"
  >
    Indentation
  </h3>
  <p
    className="card-description"
  >
    When refering to 'Indentation' in esLint this means we are talking about the stylistic way your code should look in the developer. Indentation makes your code more readable and is the first step into writing less spaghetti code. It also gives the reader visual ques for hierarchy and scoping.
  </p>
  <p>
    This is how not to do it:
  </p>
  <img
    className="category-image"
    src="https://i.imgur.com/oR17Hke.png"
  />
  <p>
    This is the proper way to do it:
  </p>
  <img
    className="category-image"
    src="https://i.imgur.com/YIp5OaH.png"
  />
  <p>
    Link for more info:
  </p>
  <a
    href="https://eslint.org/docs/rules/indent"
  >
    https://eslint.org/docs/rules/indent
  </a>
  <button
    className="return-button"
    onClick={[MockFunction]}
  >
    Finished
  </button>
</div>
`;
